cdap-adapters/src/main/java/co/cask/cdap/conversion/app/StreamConversionAdapter.java

```diff
@@ -16,17 +16,51 @@
 
 package co.cask.cdap.conversion.app;
 
-import co.cask.cdap.api.app.AbstractApplication;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.dataset.lib.FileSetProperties;
+import co.cask.cdap.api.schedule.Schedules;
+import co.cask.cdap.api.templates.AdapterConfigurer;
+import co.cask.cdap.api.templates.ApplicationTemplate;
+import com.google.gson.Gson;
+import org.apache.avro.mapreduce.AvroKeyInputFormat;
+import org.apache.avro.mapreduce.AvroKeyOutputFormat;
 
 /**
  * Application that converts a stream into a partitioned file set.
  */
-public class StreamConversionAdapter extends AbstractApplication {
+public class StreamConversionAdapter extends ApplicationTemplate<AdapterArgs> {
```

Comment:
Or is it to test the app template deployment logic. If that is the case, then ok.